.github/workflows/codeql-analysis.yml

```diff
@@ -30,7 +30,7 @@ jobs:
 
     steps:
     - name: Checkout repository
-      uses: actions/checkout@v2
```

Comment:
Choosing a specific commit instead of a tag is part of security hardening for Github Actions. https://docs.github.com/en/actions/learn-github-actions/security-hardening-for-github-actions#using-third-party-actions